mongo_connector/oplog_manager.py

```diff
@@ -419,41 +419,74 @@ def join(self):
         self.running = False
         threading.Thread.join(self)
 
-    def _pop_excluded_fields(self, doc):
+    @classmethod
+    def find_field(cls, field, doc):
+        """Find the field in the document which matches the given field.
+
+        The field may be in dot notation, eg "a.b.c". Yields a tuple
+        (path, field_value).
+        """
+        path = field.split('.')
+        try:
+            for key in path:
+                doc = doc[key]
+        except (KeyError, TypeError):
+            return
+        yield path, doc
+
+    @classmethod
+    def find_update_fields(cls, field, doc):
+        """Find the fields in the update document which match the given field.
+
+        Both the field and the top level keys in the doc may be in dot
```

Comment:
An update may or may not contain dot-notation field. For example, consider the case where `fields` is `['a.b.c']`. Updates that filter_oplog_entry should match could look like:
- `{'$set': {'a.b.c': 1}}`
- `{'$set': {'a.b': {'c':1}}}`
- `{'$set': {'a': {'b': {'c':1}}}}`